plugin/pkg/shoot/validator/admission.go

```diff
@@ -402,7 +402,7 @@ func validateProvider(c *validationContext) field.ErrorList {
 		path    = field.NewPath("spec", "provider")
 	)
 
-	if c.seed != nil {
+	if c.seed != nil && c.oldShoot.Spec.SeedName == nil && c.shoot.Spec.SeedName != nil {
```

Comment:
Can you elaborate on this check? How it is possible to determine whether it is CREATE request for a new Shoot checking by the `.spec.seedName`? At first look this check seems wrong to me as `.spec.seedName` field should no be related to the request type (`CREATE`, `UPDATE`).

Reply:
@ialidzhikov Due to the fact that the network settings are immutable, executing the network disjointedness check does only make sense exactly once at the time the shoot is assigned to a seed.

Background: After #3844, there might indeed be `Shoot`s in the system which are functional (probably only be accident) and conflict with the OpenVPN CIDR ranges. Today (with the current `master` branch) no updates to them are possible anymore. To be backwards-compatible, we decided to only execute the network disjointedness check once when the seed is initially assigned. If a shoot already has a seed then the check doesn't help anymore, though, this also fixes the mentioned problem with existing shoots.

However, there is also the control plane migration case, so probably the check could be changed to 
```suggestion
	if c.seed != nil && !apiequality.Semantic.DeepEqual(c.oldShoot.Spec.SeedName, c.shoot.Spec.SeedName) {
```
to also cover this case.

Reply:
Thank you @rfranzke, I adapted it.

Reply:
Thanks, I'm still missing unit tests:

> Please adapt the unit tests to cover your changes.

Reply:
Thanks for explaining @rfranzke .

Reply:
Unit tests are added.